### PR TITLE
Feature/gh 5386 add to cart

### PR DESCRIPTION
**Pull request by steinsebastian**

### CODE REVIEW

File: projects/core/src/occ/adapters/configurator/variant/occ-configurator.models.ts [hunk 1]
```diff
@@ -97,4 +97,8 @@ export namespace OccConfigurator {
     READ_ONLY_SINGLE_SELECTION_IMAGE = 'READ_ONLY_SINGLE_SELECTION_IMAGE',
     READ_ONLY_MULTI_SELECTION_IMAGE = 'READ_ONLY_MULTI_SELECTION_IMAGE',
   }
+
```
Comment (schuppell) [inline projects/core/src/occ/adapters/configurator/variant/occ-configurator.models.ts]: Remove type, no longer needed

File: projects/core/src/occ/adapters/configurator/variant/occ-configurator-variant.adapter.ts [hunk 2]
```diff
@@ -63,6 +65,32 @@ export class OccConfiguratorVariantAdapter
...
+    productCode: string,
+    quantity: number = 1,
+    configId: string
+  ): Observable<CartModification> {
+    const url = this.occEndpointsService.getUrl('addConfigurationToCart', {
+      userId,
+      cartId,
+    });
+
+    const occConfigOrderEntry = JSON.stringify({
```
Comment (schuppell) [inline projects/core/src/occ/adapters/configurator/variant/occ-configurator-variant.adapter.ts]: Call Serializer (needs to be created) instead

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-configuration.ts [hunk 3]
```diff
@@ -139,3 +141,13 @@ export function clickHamburger() {
...
   cy.get('cx-hamburger-menu [aria-label="Menu"]').should('be.visible');
 }
+
+export function clickAddToCartButton() {
+  cy.get(addToCartButtonSelector).click({
+    force: true,
+  });
+}
+
+export function verifyNumberOfProductsInCart(numberAsString: string) {
```
Comment (schuppell) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/product-configuration.ts]: Use checkMiniCartCount from cart.ts helper class instead

File: projects/core/src/configurator/commons/facade/configurator-commons.service.ts [hunk 4]
```diff
@@ -25,6 +34,13 @@ export class ConfiguratorCommonsService {
     );
   }
 
+  isConfigurationReady(productCode: string): Observable<boolean> {
```
Comment (schuppell) [inline projects/core/src/configurator/commons/facade/configurator-commons.service.ts]: Is this method needed, I see no consumer

File: projects/core/src/cart/facade/cart.service.ts [hunk 5]
```diff
@@ -151,6 +151,22 @@ export class CartService {
...
+      tap(cartState => {
+        if (!this.isCreated(cartState.value.content) && !cartState.loading) {
+          this.store.dispatch(
+            new CartActions.CreateCart({ userId: this.cartData.userId })
+          );
+        }
+      }),
+      filter(cartState => this.isCreated(cartState.value.content)),
+      map(cartState => cartState.value.content),
+      take(1)
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/cart/facade/cart.service.ts]: done

File: projects/core/src/configurator/commons/connectors/configurator-commons.connector.spec.ts [hunk 6]
```diff
@@ -22,6 +22,8 @@ class MockConfiguratorCommonsAdapter implements ConfiguratorCommonsAdapter {
   createConfiguration = createSpy().and.callFake(productCode =>
     of('createConfiguration' + productCode)
   );
+
+  addToCart = createSpy().and.callFake(configId => of('addToCart' + configId));
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/configurator/commons/connectors/configurator-commons.connector.spec.ts]: done

File: projects/core/src/configurator/commons/store/effects/configurator.effect.ts [hunk 7]
```diff
@@ -246,6 +251,68 @@ export class ConfiguratorEffects {
     })
   );
 
+  @Effect()
+  addToCart$: Observable<AddToCartFinalize> = this.actions$.pipe(
+    ofType(ADD_TO_CART),
+    map((action: { type: string; payload?: any }) => action.payload),
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/configurator/commons/store/effects/configurator.effect.ts]: done

File: projects/core/src/configurator/commons/store/effects/configurator.effect.ts [hunk 8]
```diff
@@ -246,6 +251,68 @@ export class ConfiguratorEffects {
...
+  @Effect()
+  addToCartFinalize$: Observable<
+    | ConfiguratorUiActions.RemoveUiState
+    | ConfiguratorActions.RemoveConfiguration
+    | CartActions.CartAddEntrySuccess
+    | CartActions.CartAddEntryFail
+  > = this.actions$.pipe(
+    ofType(ADD_TO_CART_FINALIZE),
+    map(
+      (action: {
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/configurator/commons/store/effects/configurator.effect.ts]: done

File: projects/core/src/occ/adapters/configurator/variant/converters/occ-configurator-variant-price-normalizer.spec.ts [hunk 9]
```diff
@@ -0,0 +1,57 @@
+import { Type } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/occ/adapters/configurator/variant/converters/occ-configurator-variant-price-normalizer.spec.ts]: done

File: projects/core/src/occ/adapters/configurator/variant/converters/occ-configurator-variant-price-normalizer.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/occ/adapters/configurator/variant/converters/occ-configurator-variant-price-normalizer.ts]: done

File: projects/core/src/occ/adapters/configurator/variant/occ-configurator.models.ts (hunk 1 above)
Comment (steinsebastian) [reply projects/core/src/occ/adapters/configurator/variant/occ-configurator.models.ts]: done

File: projects/core/src/configurator/commons/facade/configurator-commons.service.ts (hunk 4 above)
Comment (steinsebastian) [reply projects/core/src/configurator/commons/facade/configurator-commons.service.ts]: Not found.

File: projects/core/src/configurator/commons/store/effects/configurator.effect.ts [hunk 11]
```diff
@@ -246,6 +251,68 @@ export class ConfiguratorEffects {
...
+  @Effect()
+  addToCart$: Observable<AddToCartFinalize> = this.actions$.pipe(
+    ofType(ADD_TO_CART),
+    map((action: { type: string; payload?: any }) => action.payload),
+    switchMap(payload => {
+      return this.store.pipe(
+        select(ConfiguratorSelectors.getPendingChanges),
+        take(1),
+        filter(pendingChanges => pendingChanges === 0),
+        map(() => new AddToCartFinalize(payload))
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/configurator/commons/store/effects/configurator.effect.ts]: Done. Also changed effect groupChange/groupChangeFinalized

File: projects/core/src/model/configurator.model.ts [hunk 12]
```diff
@@ -78,4 +78,8 @@ export namespace Configurator {
     STRING = 'string',
     AUTO_COMPLETE_CUSTOM = 'input_autocomplete',
   }
+
+  export enum PriceType {
```
[the remark this answers is not in the document]
Comment (steinsebastian) [reply projects/core/src/model/configurator.model.ts]: done

File: projects/storefrontapp-e2e-cypress/cypress/helpers/product-configuration.ts (hunk 3 above)
Comment (steinsebastian) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/product-configuration.ts]: done

File: projects/core/src/occ/adapters/configurator/variant/occ-configurator-variant.adapter.ts (hunk 2 above)
Comment (steinsebastian) [reply projects/core/src/occ/adapters/configurator/variant/occ-configurator-variant.adapter.ts]: done

**Pull request merged**